### PR TITLE
Admins

### PR DESCRIPTION
**Pull request by gknisely**
@dgearhart @dnesbitt61 @kevinkreiser 

Admin updates for baldr.

### CODE REVIEW

File: valhalla/baldr/admininfo.h [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    struct Fields {
+      uint32_t name_count          :8;
+      uint32_t timezone            :6;
+      uint32_t dst                 :1;
+      uint32_t spare               :17;
+
+    } fields;
+    uint32_t value;
+  };
+
```
Comment (dnesbitt61) [inline valhalla/baldr/admininfo.h]: Admin info does NOT contain timezone or DST flag. It should contain the start/end dates for DST. Each node has to contain a timezone and dst flag (not at admin level). See: https://docs.google.com/a/mapzen.com/document/d/1ZVJ3Ptbl0BxIvg-7Xh_tY_LY88TISt7di0LkKSSnZWg/edit

File: valhalla/baldr/admininfo.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+ protected:
+
+  // Where we keep the statistics about how large the vectors below are
+  PackedItem* item_;
+
+  // List of name indexes
+  uint32_t* name_offset_list_;
+
+  // The list of names within the tile
+  const char* names_list_;
```
Comment (dgearhart) [inline valhalla/baldr/admininfo.h]: Do we have two text lists? We are not able to add names to existing text list?

**Comment by dnesbitt61** [comment]
Did we ditch the idea of storing ISO country codes and abbreviations? If we think we might want to add in the future then it should be built in now.

File: valhalla/baldr/admininfo.h (hunk 2 above)
Comment (dnesbitt61) [inline valhalla/baldr/admininfo.h]: I don't want to have separate names lists. When we add transit we can just append names and have one place that all names exist. We can de-dup incoming names against each other, but perhaps not go through the pain of de-dup against existing names

File: valhalla/baldr/admininfo.h (hunk 1 above)
Comment (gknisely) [inline valhalla/baldr/admininfo.h]: Not sure about that at admin level....for now I will leave with namecount

File: valhalla/baldr/admininfo.h (hunk 2 above)
Comment (gknisely) [inline valhalla/baldr/admininfo.h]: old code...removing.  good catch

**Comment by gknisely** [comment]
Names are saved as <name>, en:<name>, etc

**Comment by dnesbitt61** [comment]
Yes, but there will be one ISO code. If we are going to store it then add it to the base Admin structure rather than have a name lookup

File: valhalla/baldr/admininfo.h (hunk 1 above)
Comment (dnesbitt61) [inline valhalla/baldr/admininfo.h]: ?? I meant remove timezone and DST. Add fields for start and end DST date

**Comment by dgearhart** [comment]
I thought we were using the ISO 3166-1 alpha-2 for each country

File: valhalla/baldr/admininfo.h (hunk 2 above)
Comment (gknisely) [inline valhalla/baldr/admininfo.h]: sorry not old code....the admins are reusing the textlist

**Comment by dnesbitt61** [comment]
Goal is to add the fields we think we will want/need. Populate as many as we can now and leave others TODO. But that way we don't change the graph tile format each time we want to populate a field. So I think a 2 char field should be added to the base AdminInfo structure, for the ISO code of the country .

**Comment by gknisely** [comment]
So what do you want in PackedItem?  You guys are going back and forth.  
Count
Start/End dates for DST 
and what else?

**Comment by dnesbitt61** [comment]
Looks to me like admin info should contain:
Start DST
End DST
ISO code
name count
name offsets (variable size)

File: valhalla/baldr/admininfo.h [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    struct Fields {
+      uint64_t start_dst            :24; //MMDDHHMM
+      uint64_t end_dst              :24; //MMDDHHMM
+      uint64_t name_count           :8;  //name count
+      uint64_t parent_admin         :6;  //parent admin index
+      uint64_t spare                :2;
+    } fields;
+    uint64_t value;
+  };
+
```
Comment (dnesbitt61) [inline valhalla/baldr/admininfo.h]: What is the form of start and end dst and how are they converted to usable values? Wouldn't it be better to have these as separate data members outside PackedItem? Where are the access methods to read al the values (I only see name count and nameoffset(index).

**Comment by gknisely** [comment]
03080200 begins
11010200 ends
I was going add access functions next pull request for balr as I need to populate parent_admin and iso offset sooner rather than later.

**Comment by dnesbitt61** [comment]
Is that a standard way to represent a date? Seesm it would require building a string and then changing it to the numeric representation?

**Comment by gknisely** [comment]
I am trying to save space.

**Comment by dnesbitt61** [comment]
There are only a couple admin areas per tile, so the savings is a handfl of bytes. Judt seems to be a non-standard (and not trivial) conversion

**Comment by gknisely** [comment]
updated.

**Comment by dgearhart** [comment]
:+1:

**Pull request merged**